actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -97,7 +97,9 @@ def normalize_filter(filter)
           end
         end
 
-        def filter_routes(filter)
+        def filter_routes(filter, application_routes: false)
+          @routes = @routes.select { |route| route.application_set } if application_routes
```

Comment:
Good point - I kept it somewhat separate since the application_routes logic is applied before the filtering so `--application --controller Users` but would definitely make sense to move it into `filter_routes` 👍